torch/lib/TensorLib/Utils.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "TensorLib/CPUGenerator.h"
+
+namespace tlib {
+
+#define TLIB_ASSERT(cond, ...) if (!cond) { tlib::runtime_error(__VA_ARGS__); }
+
+[[noreturn]]
+void runtime_error(const char *format, ...);
+
+template <typename T, typename Base>
+static inline T* checked_cast(Base* expr, const char * name, int pos) {
```

Comment:
Because our library is "dynamic", upon an API call, we need to cast the generic "Tensor" type into the appropriate one for the function body. `checked_cast` does this, raising an exception if the cast fails.